libpod/options.go

```diff
@@ -2093,324 +2078,87 @@ func WithPodCgroup() PodCreateOption {
 
 // WithPodNoNetwork tells the pod to disable external networking.
-func WithPodNoNetwork() PodCreateOption {
-	return func(pod *Pod) error {
-		if pod.valid {
-			return define.ErrPodFinalized
-		}
-
-		if !pod.config.InfraContainer.HasInfraContainer {
-			return errors.Wrapf(define.ErrInvalidArg, "cannot disable pod networking as no infra container is being created")
-		}
-
-		if len(pod.config.InfraContainer.PortBindings) > 0 ||
-			pod.config.InfraContainer.StaticIP != nil ||
-			pod.config.InfraContainer.StaticMAC != nil ||
-			len(pod.config.InfraContainer.Networks) > 0 ||
-			pod.config.InfraContainer.HostNetwork {
-			return errors.Wrapf(define.ErrInvalidArg, "cannot disable pod network if network-related configuration is specified")
-		}
-
-		pod.config.InfraContainer.NoNetwork = true
-
-		return nil
+func WithPodNoNetwork(infraSpec *specgen.SpecGenerator) error {
+	if len(infraSpec.PortMappings) > 0 ||
+		infraSpec.StaticIP != nil ||
+		infraSpec.StaticMAC != nil ||
+		len(infraSpec.CNINetworks) > 0 ||
+		infraSpec.NetNS.NSMode == "host" {
+		return errors.Wrapf(define.ErrInvalidArg, "cannot disable pod network if network-related configuration is specified")
 	}
+	infraSpec.NetNS.NSMode = specgen.NoNetwork
+	return nil
 }
 
 // WithPodHostNetwork tells the pod to use the host's network namespace.
-func WithPodHostNetwork() PodCreateOption {
-	return func(pod *Pod) error {
-		if pod.valid {
-			return define.ErrPodFinalized
-		}
-		if !pod.config.InfraContainer.HasInfraContainer {
-			return errors.Wrapf(define.ErrInvalidArg, "cannot configure pod host networking as no infra container is being created")
-		}
-
-		if len(pod.config.InfraContainer.PortBindings) > 0 ||
-			pod.config.InfraContainer.StaticIP != nil ||
-			pod.config.InfraContainer.StaticMAC != nil ||
-			len(pod.config.InfraContainer.Networks) > 0 ||
-			pod.config.InfraContainer.NoNetwork {
-			return errors.Wrapf(define.ErrInvalidArg, "cannot set host network if network-related configuration is specified")
-		}
-
-		pod.config.InfraContainer.HostNetwork = true
-
-		return nil
+func WithPodHostNetwork(infraSpec *specgen.SpecGenerator) error {
+	if len(infraSpec.PortMappings) > 0 ||
+		infraSpec.StaticIP != nil ||
+		infraSpec.StaticMAC != nil ||
+		len(infraSpec.CNINetworks) > 0 ||
+		infraSpec.NetNS.NSMode == specgen.NoNetwork {
+		return errors.Wrapf(define.ErrInvalidArg, "cannot set host network if network-related configuration is specified")
 	}
+	infraSpec.NetNS.NSMode = specgen.Host
+	return nil
 }
 
 // WithPodInfraExitCommand sets an exit command for the pod's infra container.
 // Semantics are identical to WithExitCommand() above - the ID of the container
 // will be appended to the end of the provided command (note that this will
 // specifically be the ID of the infra container *and not the pod's id*.
-func WithPodInfraExitCommand(exitCmd []string) PodCreateOption {
+func WithPodInfraExitCommand(exitCmd []string, infra *specgen.SpecGenerator) PodCreateOption {
```

Comment:
Remove

Reply:
moved content to pod_create 

``
		exitCommand, err := CreateExitCommandArgs(storageConfig, runtimeConfig, logrus.IsLevelEnabled(logrus.DebugLevel), false, false)
		if err != nil {
			return nil, errors.Wrapf(err, "error creating infra container exit command")
		}
		p.InfraContainerSpec.InfraExitCommand = exitCommand
``